pkg/kubelet/container_manager_linux.go

```diff
@@ -0,0 +1,120 @@
+// +build linux
+
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubelet
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util/errors"
+	"github.com/docker/libcontainer/cgroups"
+	"github.com/docker/libcontainer/cgroups/fs"
+	"github.com/docker/libcontainer/configs"
+	"github.com/golang/glog"
+)
+
+type containerManagerImpl struct {
+	// Absolute name of the desired container that Docker should be in.
+	dockerContainerName string
+
+	// The manager of the resource-only container Docker should be in.
+	manager fs.Manager
+}
+
+var _ containerManager = &containerManagerImpl{}
+
+// Takes the absolute name that the Docker daemon should be in.
+// Empty container name disables moving the Docker daemon.
+func newContainerManager(dockerDaemonContainer string) (containerManager, error) {
+	return &containerManagerImpl{
+		dockerContainerName: dockerDaemonContainer,
+		manager: fs.Manager{
+			Cgroups: &configs.Cgroup{
+				Name:            dockerDaemonContainer,
+				AllowAllDevices: true,
+			},
+		},
+	}, nil
+}
+
+func (cm *containerManagerImpl) Start() error {
```

Comment:
Why ensure in a loop? Can't we move it once and check on restart? This is extra work when we know noone is moving them around.